internal/database/memory/database.go

```diff
@@ -0,0 +1,96 @@
+// Package memory provides an in-memory database implementation.
+package memory
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ChainSafe/gossamer/internal/database"
+)
+
+// Database is an in-memory database implementation.
+type Database struct {
```

Comment:
The point of the in-memory database implementation package is to force the internal/database code to be flexible for additional implementations.

On top of that, I would suggest us to use this one instead of the in-memory badger database, so the badger database is only on disk.

Reply:
It could if we add a config field for the database implementation later.
Not that useful I have to admit, but I would say it serves as a good example for another database implementation (as in it's a 'dumb' base just using Go std types).

Reply:
if tests can use the badger in memory version, do we really need this in memory implementation?

Reply:
I would like to keep it as a base for another future implementation. For example `badger` uses the badger/v3 batch implementation, but some other implementations might need to implement their own batching, and it's already there in the (zero-dependency) memory implementation. I can create an issue to remove it later on once we have at least another implementation?